hs_tools_resource/page_processors.py

```diff
@@ -33,23 +33,12 @@ def landing_page(request, page):
                                     element_id=content_model.metadata.url_bases.first().id
                                     if content_model.metadata.url_bases.first() else None)
 
-        ResTypeFormSetEdit = formset_factory(wraps(ResTypeForm)(partial(ResTypeForm,
-                                                                        allow_edit=edit_resource)),
-                                             formset=BaseFormSet, extra=0)
-        res_type_formset = ResTypeFormSetEdit(initial=content_model.metadata.res_types.all().values(),
-                                              prefix='res_type')
-        add_toolresourcetype_modal_form = ResTypeForm(allow_edit=edit_resource, res_short_id=content_model.short_id)
-
-        for form in res_type_formset.forms:
-            if len(form.initial) > 0:
-                form.delete_modal_form = MetaDataElementDeleteForm(content_model.short_id,
-                                                                   'toolresourcetype',
-                                                                   form.initial['id'])
-                form.action = "/hsapi/_internal/%s/toolresourcetype/%s/update-metadata/" % (content_model.short_id,
-                                                                                            form.initial['id'])
-                form.number = form.initial['id']
-            else:
-                form.action = "/hsapi/_internal/%s/toolresourcetype/add-metadata/" % content_model.short_id
+        res_type_form = ResTypeForm(instance=content_model.metadata.res_types.first(),
+                                    res_short_id=content_model.short_id,
+                                    element_id=content_model.metadata.res_types.first().id
+                                    if content_model.metadata.res_types.first() else None)
```

Comment:
this has nothing to do with crispy forms.

In a single expression between lines 36 and 39, you are calling `content_model.metadata.res_types.first()` three separate times. You should be calling it once first, storing the result in a local variable, and using that instead of the three separate calls.